src/analytics/SegmentProvider.tsx

```diff
@@ -32,21 +32,28 @@ function getAdobeVisitorId() {
   return -1;
 }
 
-const getAPIKey = (env: SegmentEnvs = 'dev', module: SegmentModules, moduleAPIKey?: string) =>
-  moduleAPIKey ||
-  {
-    prod: {
-      acs: '9NmgZh57uEaOW9ePKqeKjjUKE8MEqaVU',
-      hacCore: 'cLLG3VVakAECyGRAUnmjRkSqGJkYlRWI',
-      openshift: 'z3Ic4EtzJtHrhXfpKgViJmf2QurSxXb9',
-    },
-    dev: {
-      acs: 'CA5jdEouFKAxwGq7X9i1b7UySMKshj1j',
-      hacCore: '5SuWCF4fRqTzMD8HVsk2r1LEYsYVsHCC',
-      openshift: 'A8iCO9n9Ax9ObvHBgz4hMC9htKB0AdKj',
-    },
-  }[env]?.[module] ||
-  KEY_FALLBACK[env];
+const getAPIKey = (env: SegmentEnvs = 'dev', module: SegmentModules, moduleAPIKey?: string, moduleAPIKeyDev?: string) => {
+  // Use the appropriate key based on environment
+  const envSpecificKey = env === 'prod' ? moduleAPIKey : moduleAPIKeyDev;
```

Comment:
@CodyWMitchell we should investigate adding some testing in this area. It would be easy for regressions in this new logic to go un-noticed.